src/canvasMachine.tsx

```diff
@@ -186,32 +186,39 @@ export const canvasMachine = canvasModel.createMachine({
       }),
     },
     'elkGraph.UPDATE': {
-      actions: canvasModel.assign({
-        elkGraph: (_, e) => e.elkGraph,
-      }),
+      actions: [
+        canvasModel.assign({
+          elkGraph: (_, e) => e.elkGraph,
+        }),
+        send('FIT_TO_VIEW'),
+      ],
     },
     FIT_TO_VIEW: {
       actions: [
         canvasModel.assign({
           zoom: (ctx) => {
+            if (!ctx.elkGraph) return ctx.zoom;
```

Comment:
why do we need this?

Reply:
Because `ctx.elkGraph` can be undefined, and if it is we need to do nothing, silently